cmd/root.go

```diff
@@ -25,3 +28,15 @@ func Execute() error {
 func RootCmd() *cobra.Command {
 	return rootCmd
 }
+
+// DefaultCommandRunFunction default run function logic
+func DefaultCommandRunFunction(service KoolService) CobraRunFN {
```

Comment:
This is it! nice.

Now, I am not sure how many of our commands will get this DefaultCommandRunFunction, but maybe one of them already have a special situation (the `restart`) where we have a list of KoolService... so maybe this could already be generic, receiving `DefaultCommandRunFunction(service KoolService...)`, and then we do the same execution thing, one by one, in a loop, failing in the first error.

Reply:
Also, a great thing for this is, we should write a test for this generic function based on KoolService, passing FakeKoolServices, and then we test this whole thing only once! and tests for these default commands become a bit less redundant/repetitive.